### PR TITLE
#359 Start nightlies on repository dispatch

### LINKED ISSUE #359: Start nightlies on repository dispatch

### PR DESCRIPTION
**Pull request by valleefred**
Closes #359 

**Checklist before requesting a review**

*use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes*

- [ ] unit tests and non regression tests were added (update of algorithms)
- [ ] main documentation was updated (update of input/output file, update of algorithms)
- [x] the corresponding milestone was added in the ticket and in this PR
- [ ] if this PR modifies a dictionary: the corresponding french dictionary was updated

### CODE REVIEW

File: .github/workflows/nightly.yml [hunk 1]
```diff
@@ -19,19 +24,13 @@ jobs:
...
     steps:
       - name: Checkout sources
-        uses: actions/checkout@v1
+        uses: actions/checkout@v3
 
       - name: Fetch Dynawo
-        run: |
-          source .github/workflows/tools.sh
-          download_asset $GH_API_DYNAWO nightly Dynawo_omc_v${DYNAWO_VERSION}.zip $GITHUB_TOKEN
+        run: curl -LOs https://github.com/dynawo/dynawo/releases/download/nightly/Dynawo_omc_v$DYNAWO_VERSION.zip
```
Comment (gautierbureau) [inline .github/workflows/nightly.yml:33]: Use variables to determine the urls, it is useful when forking

**Review by gautierbureau — CHANGES_REQUESTED** [review]

File: .github/workflows/nightly.yml (hunk 1 above)
Comment (gautierbureau) [reply .github/workflows/nightly.yml:33]: also keep the tokens it avoid having issues with api request limits

Comment (valleefred) [reply .github/workflows/nightly.yml]: Use ${{ github.repository_owner }} in place of first dynawo in url.
Token is not needed to get assets from a public repo.

File: .github/workflows/nightly.yml [hunk 2]
```diff
@@ -2,8 +2,13 @@ name: Nightly
 
 on:
   workflow_dispatch:
-  schedule:
-    - cron:  '0 1 * * 1-5'
+  repository_dispatch:
+    types: [nightly-linux]
+  # schedule:
```
Comment (gautierbureau) [inline .github/workflows/nightly.yml:7]: remove this

**Review by gautierbureau — CHANGES_REQUESTED** [review]

File: .github/workflows/nightly.yml (hunk 2 above)
Comment (valleefred) [reply .github/workflows/nightly.yml:7]: OK remove schedule

**Review by gautierbureau — APPROVED** [review]

**Pull request merged**